[SaferCpp] Address issues in FindController.cpp

kmonsen opened the pull request:
#### b5047f08e4dab0fa10feaa440e68a6e6de938577
<pre>
[SaferCpp] Address issues in FindController.cpp
<a href="https://rdar.apple.com/148773692">rdar://148773692</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=291230">https://bugs.webkit.org/show_bug.cgi?id=291230</a>

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::mainFramePlugIn):
(WebKit::FindController::countStringMatches):
(WebKit::FindController::replaceMatches):
(WebKit::FindController::updateFindUIAfterPageScroll):
(WebKit::FindController::findStringIncludingImages):
(WebKit::FindController::findString):
(WebKit::FindController::findStringMatches):
(WebKit::FindController::findRectsForStringMatches):
(WebKit::FindController::indicateFindMatch):
(WebKit::FindController::hideFindUI):
(WebKit::FindController::updateFindIndicator):
(WebKit::FindController::didFindString):
(WebKit::FindController::rectsForTextMatchesInRect):
(WebKit::FindController::drawRect):

Canonical link: <a href="https://commits.webkit.org/293446@main">https://commits.webkit.org/293446@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/07dffd20432cff36b51843d793b70254b91beed2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98948 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18585 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8829 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104076 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49539 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18880 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27035 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75324 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32451 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101952 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14345 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89362 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55685 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14136 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48917 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84071 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7410 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106443 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26045 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18984 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84286 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26420 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85559 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83788 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21241 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28444 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6118 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/19767 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25998 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31184 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25818 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29138 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27392 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/WebPage/FindController.cpp:
```diff
@@ -78,7 +78,8 @@ FindController::~FindController()
 PluginView* FindController::mainFramePlugIn()
 {
-    return m_webPage->mainFramePlugIn();
+    RefPtr protectedWebPage { m_webPage.get() };
```
We do not use `protected` naming for local variables in new code. This should just be `webPage`

cdumez commented on Source/WebKit/WebProcess/WebPage/FindController.cpp:
```diff
@@ -78,7 +78,8 @@ FindController::~FindController()
 {
-    return m_webPage->mainFramePlugIn();
+    RefPtr protectedWebPage { m_webPage.get() };
+    return protectedWebPage->mainFramePlugIn();
```
It seems there an are a few places in this PR where we could avoid extra variables by introducing a new `protectedWebPage()` member function so I think we should do that. Local variables are normally used to avoid repeated calls to `protectedFoo()` functions within the same function.

kmonsen committed:
[SaferCpp] Address issues in FindController.cpp
rdar://148773692
https://bugs.webkit.org/show_bug.cgi?id=291230

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::mainFramePlugIn):
(WebKit::FindController::countStringMatches):
(WebKit::FindController::replaceMatches):
(WebKit::FindController::updateFindUIAfterPageScroll):
(WebKit::FindController::findStringIncludingImages):
(WebKit::FindController::findString):
(WebKit::FindController::findStringMatches):
(WebKit::FindController::findRectsForStringMatches):
(WebKit::FindController::indicateFindMatch):
(WebKit::FindController::hideFindUI):
(WebKit::FindController::updateFindIndicator):
(WebKit::FindController::didFindString):
(WebKit::FindController::rectsForTextMatchesInRect):
(WebKit::FindController::drawRect):

Canonical link: https://commits.webkit.org/293446@main
